### PR TITLE
update validation more than on tokenfeeder for the same token

### PR DESCRIPTION
**Pull request by leonz789**
## Description

<p> In the process of params validation, when there're more than one `tokenfeeders` for the same token, check for if they are overlapped or have continuous roundID </p>
----

Closes #XXX

## Summary by CodeRabbit

- **New Features**
  - Introduced enhanced validation checks for the TokenFeeder functionality, ensuring parameters are consistently valid and preventing overlapping configurations.
  
- **Bug Fixes**
  - Improved the reliability of the test suite for TokenFeeder, including renamed tests and added validation tests to cover various scenarios and edge cases.

### CODE REVIEW

**Comment by leonz789** [comment]
resolved #138

File: x/oracle/keeper/params_test.go [hunk 1]
```diff
@@ -215,3 +215,75 @@ func TestUpdateTokenFeeder(t *testing.T) {
 		})
 	}
 }
+
+func TestTokenFeederValidate(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (leonz789) [reply x/oracle/keeper/params_test.go:226]: minimum interval is 1

**Review by mikebraver — APPROVED** [review]

File: x/oracle/types/params.go [hunk 2]
```diff
@@ -152,8 +153,26 @@ func (p Params) Validate() error {
...
+			// this is one feeder that succeeded the previous one, so we should check the roundID is continuoslly increased
+			prevEndRoundID := prev.StartRoundID + (prev.EndBlock-prev.StartBaseBlock)/prev.Interval
+			if feeder.StartRoundID != prevEndRoundID+1 {
+				return ErrInvalidParams.Wrap("invalid tokenFeeder, for the same token the roundID should be continuous")
+			}
+		}
+		feeders[feeder.TokenID] = feeder
 	}
-
+	//}
```
Comment (bwhour) [inline x/oracle/types/params.go:176]: remove this line.

**Review by bwhour — APPROVED** [review]

**Review by TimmyExogenous — APPROVED** [review]

**Comment by cloud8little** [comment]
Test passed. Sample genesis file for token feeder.
 ```
         {
            "token_id": "1",
            "rule_id": "1",
            "start_round_id": "1",
            "start_base_block": "30",
            "interval": "10",
            "end_block": "0"
          },
          {
            "token_id": "1",
            "rule_id": "1",
            "start_round_id": "1",
            "start_base_block": "10000000",
            "interval": "10",
            "end_block": "0"
          }
```

```
exocored validate-genesis --home /root/.test-pr146
Error: error validating genesis file /root/.test-pr146/config/genesis.json: invalid tokenFeeder, for the same token at most one running or future tokenFeeder without endblock set: invalid params
```

**Comment by cloud8little** [comment]
@leonz789 plz resolve conflicts for UT file.

**Review by cloud8little — APPROVED** [review]

**Review by MaxMustermann2 — APPROVED** [review]
